### PR TITLE
ButtonTextWithAddIcon shared component

### PR DESCRIPTION
**Pull request by flyinghermit**
The add button with icon is implemented among various components so this PR abstracts the component to shared component. 

Screenshot 2023-12-28 at 5 28 43 PM

Screenshot 2023-12-28 at 5 26 55 PM

Also updates `LabelCreater` component to use `ButtonTextWithAddIcon`

### CODE REVIEW

File: web/packages/shared/components/ButtonTextWithAddIcon/ButtonTextWithAddIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      `}
+      disabled={disabled}
+    >
+      <AddIcon
+        className="icon-add"
+        size={iconSize}
+        css={`
+          margin-right: 3px;
+        `}
+        disabled={disabled}
```
Comment (ravicious) [inline web/packages/shared/components/ButtonTextWithAddIcon/ButtonTextWithAddIcon.tsx:54]: This `disabled` ends up being added on `<span>` which has no effect. I think it's a mistake made when the new `Icon` component was introduced. We should be able to safely remove `disabled` from here.

**Review by ravicious — APPROVED** [review]
Approved with some minor comments.

File: web/packages/shared/components/ButtonTextWithAddIcon/ButtonTextWithAddIcon.story.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        disabled={true}
+      />
+      <ButtonTextWithAddIcon
+        label={'Add Item Disabled with Medium Icon Size'}
+        onClick={() => null}
+        iconSize={'medium'}
+      />
+      {/* <ButtonTextWithAddIcon mt={3} title="Barnes" ssoType="google" />
+    <ButtonTextWithAddIcon mt={3} title="Norton" ssoType="bitbucket" />
+    <ButtonTextWithAddIcon mt={3} title="Russell" ssoType="unknown" /> */}
```
[the remark this answers is not in the document]
Comment (flyinghermit) [reply web/packages/shared/components/ButtonTextWithAddIcon/ButtonTextWithAddIcon.story.tsx]: sorry for unpolished story! addressed this and above comment in https://github.com/gravitational/teleport/pull/36103/commits/50c64d270c894d8c2c8a4f71bd3680a2132ef3dc

File: web/packages/shared/components/ButtonTextWithAddIcon/ButtonTextWithAddIcon.tsx (hunk 1 above)
Comment (flyinghermit) [reply web/packages/shared/components/ButtonTextWithAddIcon/ButtonTextWithAddIcon.tsx]: I just verified this. You are correct. Thanks for taking a deeper look :)
https://github.com/gravitational/teleport/pull/36103/commits/a7b8d57c2652a1f85d96049f3eeb2e726acf607c

File: web/packages/teleport/src/Discover/Shared/LabelsCreater/LabelsCreater.tsx [hunk 3]
```diff
@@ -174,30 +175,15 @@ export function LabelsCreater({
...
+      <ButtonTextWithAddIcon
         css={`
-          padding-left: 0px;
-          &:disabled {
-            .icon-add {
-              opacity: 0.35;
-            }
-            pointer-events: none;
-          }
+          margin-top: -2px;
```
Comment (kimlisa) [inline web/packages/teleport/src/Discover/Shared/LabelsCreater/LabelsCreater.tsx:180]: i don't think this is respected, i'd remove it. it was to push up the plus icon so it's more aligned, but it's so slight you can't really tell

**Review by kimlisa — APPROVED** [review]

File: web/packages/teleport/src/Discover/Shared/LabelsCreater/LabelsCreater.tsx (hunk 3 above)
Comment (flyinghermit) [reply web/packages/teleport/src/Discover/Shared/LabelsCreater/LabelsCreater.tsx]: removed in https://github.com/gravitational/teleport/pull/36103/commits/afc2f96d25910bae3538d5102ec04725baab8bcf

**Pull request merged**